oak/proto/permissions.proto

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+package oak.permissions;
+
+// A PermissionsConfiguration provides a declarative description of the features that are permitted
+// for an Oak application.
+//
+// The permissions are applied dynamically at runtime, when creating the Nodes.
+message PermissionsConfiguration {
+  // Whether to enable the use of a GrpcServerNode as the front-door to the application.
+  bool grpc_front_door = 1;
+
+  // Whether to enable the use of an HttpServerNode as the front-door to the application.
+  bool http_front_door = 2;
+
+  // Whether to allow application logging. If true, allows creating a LogNode for the application.
+  bool logging = 3;
+
+  // Whether to allow monitoring (for the Runtime and the application). If true, the monitoring
+  // auxiliary service will be started.
+  bool monitoring = 4;
+
+  // Whether to allow introspection. If true, the introspection auxiliary service will be started.
+  bool introspection = 5;
+
+  // List of the allowed domain names or IP addresses that the application can connect to. Only
+  // GrpcClientNodes that are configured to connect to servers in this list can be created.
+  repeated string grpc_client_domains = 6;
+
+  // List of allowed roughtime servers that the application can connect to.
+  repeated string roughtime_servers = 7;
```

Comment:
This is potentially problematic. This could allow the owner of the site to specify only a single roughtime server (perhaps one they control), which would mean that the roughtime loses a lot of the trust related to the multi-server approach.

Reply:
I think the idea is that this config is static, and publicly available. So it can be reviewed, and people will know that an app with such a config cannot be trusted. Does that make it better?

Reply:
We should not assume that any remote client would review this config (hence why it is a defense in depth approach). And as @conradgrobler mentioned, we should not let whoever is running the runtime select the list runtime servers that are allowed; the list of servers should be hardcoded in the implementation of the runtime itself (which makes it part of the TCB and the remotely attestable data).

Reply:
Removed, as discussed.